Oraimunka24015004/src/AlgorithmicRecap.java

```diff
@@ -0,0 +1,172 @@
+            if (map.containsKey(character)) {
+                map.put(character, map.get(character) + 1);
+            } else {
+                map.put(character, 1);
+            }
+        }
+        return map;
+    }
+
+    BigInteger taskFourCountNumbers(String text) {
+        BigInteger result = BigInteger.valueOf(0);
+        for (int i = 0; i < text.length(); i++) {
+            Character character = text.charAt(i);
+            if (Character.isDigit(character)) {
+                result = result.add(BigInteger.valueOf(Character.getNumericValue(character)));
+            }
+        }
+        return result;
+    }
+
+    int taskFiveGetFactorial(int number) {
+        int result = 1;
+        for (int i = 1; i <= number; i++) {
+            result *= i;
+        }
+        return result;
+    }
+
+    char taskSixGetUniqueChar(String text) {
+        Map<Integer, Character> map = new HashMap<>();
+        int key = 0;
+        char result = ' ';
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            if (map.containsValue(character)) {
+                map.put(key + 1, character);
+            } else {
+                map.put(key + 1, character);
+                result = character;
+            }
+        }
+        return result;
+    }
+
+    String taskSevenGetAllUniqueChar(String text) {
+        String result = "";
+        List<Character> blacklist = new ArrayList<>();
+        int counter = 0;
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            for (int j = i + 1; j < text.length(); j++) {
+                char tempChar = text.charAt(j);
+                if (tempChar == character) {
+                    counter++;
+                }
+            }
+            if (counter == 0 && !blacklist.contains(character)) {
+                result += character;
+            } else {
+                blacklist.add(character);
+                counter = 0;
+            }
+        }
+        return result;
+    }
+
+    boolean taskEightSameAmountOfAb(String text, String text2) {
+        return (countSimilarChar('a', text) == countSimilarChar('a', text2))
+                && countSimilarChar('b', text) == countSimilarChar('b', text2);
+    }
+
+    private int countSimilarChar(char getChar, String text) {
+        int counter = 0;
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            if (character == getChar) {
+                counter++;
+            }
+        }
+        return counter;
+    }
+
+    boolean taskNineArrayContains(Integer[] numbers, Integer number) {
+        List<Integer> list = Arrays.asList(numbers);
+        Set<Integer> set = new HashSet<>(list);
+        if (set.contains(number)) {
+            return true;
+        }
+        return false;
+    }
+
+    StringBuilder taskTenWhichString(List<String> texts, char character) {
```

Comment:
never return a StringBuilder. Also, what is this task name? no idea what it is doing based on the name. `taskOne`, etc. prefixes are also completely unnecessary. please remove them from everywhere

Reply:
better to return in a list though